lib/TH/generic/THStorage.h

```diff
@@ -28,6 +29,8 @@ typedef struct THStorage
     char flag;
     THAllocator *allocator;
     void *allocatorContext;
+    int type;
```

Comment:
So given a pointer to a THStorage, how would you know it is float or double, etc?

  THStorage *storage = luaT_checkudata(L, 1, torch_Storage);
  // what type of data is in storage?